### ISSUE TITLE
undefined method `first' for #<Pry::Prompt:> (NoMethodError)

### ISSUE THREAD

**Issue by teampolyglot** [ascendbruce/awesome_rails_console#11]
My `Gemfile.lock` has this:

```
    awesome_rails_console (0.4.4)
      awesome_print
      pry-rails
      railties
...
    pry (0.13.0)
      coderay (~> 1.1)
      method_source (~> 1.0)
    pry-rails (0.3.9)
      pry (>= 0.10.4)
```

Running `rails console` fails with this error:

```
Loading development environment (Rails 6.0.2.2)
warning: setting prompt with help of `Pry.config.prompt = [proc {}, proc {}]` is deprecated. Use Pry::Prompt API instead
Traceback (most recent call last):
        32: from bin/rails:4:in `<main>'
        31: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.0.2.2/lib/active_support/dependencies.rb:325:in `require'
        30: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.0.2.2/lib/active_support/dependencies.rb:291:in `load_dependency'
        29: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.0.2.2/lib/active_support/dependencies.rb:325:in `block in require'
        28: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
        27: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
        26: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
        25: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
        24: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
        23: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.0.2.2/lib/rails/commands.rb:18:in `<main>'
        22: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.0.2.2/lib/rails/command.rb:46:in `invoke'
        21: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.0.2.2/lib/rails/command/base.rb:69:in `perform'
        20: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
        19: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
        18: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
        17: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.0.2.2/lib/rails/commands/console/console_command.rb:102:in `perform'
        16: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.0.2.2/lib/rails/commands/console/console_command.rb:19:in `start'
        15: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.0.2.2/lib/rails/commands/console/console_command.rb:70:in `start'
        14: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/pry_class.rb:191:in `start'
        13: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:15:in `start'
        12: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:38:in `start'
        11: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/input_lock.rb:78:in `with_ownership'
        10: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/input_lock.rb:61:in `__with_ownership'
         9: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:38:in `block in start'
         8: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:67:in `repl'
         7: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:67:in `loop'
         6: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:68:in `block in repl'
         5: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/repl.rb:95:in `read'
         4: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/pry_instance.rb:471:in `select_prompt'
         3: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/pry_class.rb:382:in `critical_section'
         2: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/pry_instance.rb:490:in `block in select_prompt'
         1: from /Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/pry-0.13.0/lib/pry/pry_instance.rb:692:in `generate_prompt'
/Users/helix/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/awesome_rails_console-0.4.4/lib/awesome_rails_console/railtie.rb:37:in `block in show_rails_env_name_before_prompt': undefined method `first' for #<Pry::Prompt:0x00007fc2179ae1c0> (NoMethodError)
```

**Comment by pharapeti** [comment]
I can also reproduce this error

**Comment by sebyx07** [comment]
add these to gemfile

```
  gem 'pry', '~> 0.12.2'
  gem 'pry-byebug', '~> 3.7'
```

in `group :development, :test do`

I think it's because of a newer version of pry

**Comment by pharapeti** [comment]
> add these to gemfile
> 
> ```
>   gem 'pry', '~> 0.12.2'
>   gem 'pry-byebug', '~> 3.7'
> ```
> 
> in `group :development, :test do`
> 
> I think it's because of a newer version of pry

Works for me

**Comment by deXterbed** [comment]
> add these to gemfile
> 
> ```
>   gem 'pry', '~> 0.12.2'
>   gem 'pry-byebug', '~> 3.7'
> ```
> 
> in `group :development, :test do`
> 
> I think it's because of a newer version of pry

doesn't work for rails 6.0.3

```
Bundler could not find compatible versions for gem "method_source":
  In snapshot (Gemfile.lock):
    method_source (= 1.0.0)

  In Gemfile:
    pry (~> 0.12.2) was resolved to 0.12.2, which depends on
      method_source (~> 0.9.0)

    rails (~> 6.0.3) was resolved to 6.0.3, which depends on
      railties (= 6.0.3) was resolved to 6.0.3, which depends on
        method_source
```

**Comment by Inkybro** [comment]
Seems to work for me on Rails 6.0.3.2, but there are some warnings:

```
Running via Spring preloader in process 3145
Loading development environment (Rails 6.0.3.2)
/home/inky/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/pry-byebug-3.8.0/lib/pry-byebug/pry_ext.rb:13: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/home/inky/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/pry-stack_explorer-0.4.9.3/lib/pry-stack_explorer/when_started_hook.rb:63: warning: in `eval'
/home/inky/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/railties-6.0.3.2/lib/rails/commands/console/console_command.rb:70: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/home/inky/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/pry-stack_explorer-0.4.9.3/lib/pry-stack_explorer/when_started_hook.rb:63: warning: in `eval'

Frame number: 0/24
<main>: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/home/inky/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/pry-0.12.2/lib/pry/commands/whereami.rb:40: warning: in `eval'
<main>: warning: __LINE__ in eval may not return location in binding; use Binding#source_location instead
/home/inky/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/pry-0.12.2/lib/pry/commands/whereami.rb:41: warning: in `eval'
Development [1] ngo(main)> exit
```

**Comment by linchiquei** [comment]
> > add these to gemfile
> > ```
> >   gem 'pry', '~> 0.12.2'
> >   gem 'pry-byebug', '~> 3.7'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > in `group :development, :test do`
> > I think it's because of a newer version of pry
> 
> Works for me

it work for me too ! thanks !!! 
my rails version is 6.1.4.1
gemfile look like this
`group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
  gem 'pry', '~> 0.12.2'
  gem 'awesome_rails_console'
end
.....
group :development, :test do
  gem 'hirb'
  gem 'hirb-unicode-steakknife', require: 'hirb-unicode'
  gem 'pry-byebug', '~> 3.7'
  gem 'pry-stack_explorer'
end
`

**Comment by ascendbruce** [comment]
This should be fixed in version `0.4.5`
